### ISSUE TITLE
TS0601_temperature_humidity_sensor not working

### ISSUE THREAD

**Issue by gemo-bi** [ioBroker/ioBroker.zigbee#1683]
Hello,
unfortunately, my thermometers do not deliver any values.
Please can you help me!

Zigbee 1.18.10, Nodejs 18.14.0, ConBee2 Stick
Sensor:
https://www.zigbee2mqtt.io/devices/TS0601_temperature_humidity_sensor.html

![Screenshot 2023-02-14 184823](https://user-images.githubusercontent.com/62117738/218818596-a582c880-6433-44b0-afe0-ab0857171a03.png)
![Screenshot 2023-02-14 184910](https://user-images.githubusercontent.com/62117738/218818598-a2ade074-0884-4065-b579-70ec0ad2ef3c.png)
![Screenshot 2023-02-14 185113](https://user-images.githubusercontent.com/62117738/218818599-a116e4da-197e-47ad-add1-e1b986485be2.png)

**Comment by asgothian** [comment]
Please post the info on the info field (the blue (i) on the tile background. You seem to have an unsupported variant of the device.

**Comment by gemo-bi** [comment]
![2023-02-15 06_00_03](https://user-images.githubusercontent.com/62117738/218936224-8bc23de6-fcf8-4996-9726-71ce568196f0.png)

**Comment by asgothian** [comment]
🤦🏾 

You cut out the important part of the information panel. Please post the entire panel

**Comment by asgothian** [comment]
Please remove the device and pair it again.

A.

**Comment by gemo-bi** [comment]
Devise has been removed and paired again, but nothing has changed.

Blue Info:
model:TS0601_temperature_humidity_sensor
description:Temperature & humidity sensor
modelZigbee:TS0601
type:EndDevice
nwk:16631
manuf id:4417
manufacturer:_TZE200_bjawzodf
power:Battery
app version:70
hard version:1
zcl version:3
stack version:0
date code:
interviewed:true
configured:false
endpoint:1
profile:260
input clusters:genGroups (4)
genScenes (5)
manuSpecificTuya (61184)
genBasic (0)
output clusters:genOta (25)
genTime (10)
![Screenshot 2023-02-16 055842](https://user-images.githubusercontent.com/62117738/219272683-5ea72e8c-d658-41f3-bdb2-8d083ab51312.png)
![Screenshot 2023-02-16 055908](https://user-images.githubusercontent.com/62117738/219272685-ba7c7c9c-e9b6-446f-b8e6-a4d12c2ddf7f.png)

**Comment by asgothian** [comment]
Just because we have been chasing a ghost for another problem:

Can you verify that the 1.8.10 version of the adapter is running ? The easiest way to do that is to restart the adapter and check in the iobroker log which adapter is listed as running. There have been situations where the Adapter was not updated properly, usually in combination with master/slave setups. The Integration of the device is present and clean - it seems that the zigbee subsystem detects the correct device ,but the iobroker does not generate the exposed states.

A.

**Comment by gemo-bi** [comment]
zigbee.0
2023-02-16 12:20:56.490	info	starting. Version 1.8.10 (non-npm: ioBroker/ioBroker.zigbee#1dc42c491b88df766160071889c06e88914838d7) in /opt/iobroker/node_modules/iobroker.zigbee, node: v18.14.0, js-controller: 4.0.24

**Comment by gemo-bi** [comment]
Hi, is there also a solution for my problem?

**Comment by asgothian** [comment]
Possibly. Please add the IEEE of your device to the state zigbee.0.info.debugmessages and leave the device for about 60 minutes. Afterwards, check the log for WARN messages with the keyword "elevated" and post these.

The integration of the device seems correct, the adapter version seems correct, which leaves either the device or another internal error.

A.

**Comment by gemo-bi** [comment]
hello, unfortunately there are no log entries with "elevated".
I have now also updated the firmware of my Conbee2, but hasn't changed anything

I have set the IEEE in the debugmessage, i hope this is correct ...
![Screenshot 2023-02-19 114943](https://user-images.githubusercontent.com/62117738/219943504-dbabcbf9-63ac-410e-8211-c0d00b1b50c4.png)
[iob_log.txt](https://github.com/ioBroker/ioBroker.zigbee/files/10776665/iob_log.txt)

**Comment by gemo-bi** [comment]
found this in the log....

2023-02-19 11:54:31.247 - error: zigbee.0 (832) DeviceConfigure:Failed to DeviceConfigure.configure 0xa4c13892e9c0e962 TS0601 (TypeError: mappedDevice.configure is not a function
at DeviceConfigure.doConfigure (/opt/iobroker/node_modules/iobroker.zigbee/lib/zbDeviceConfigure.js:134:32)
at processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Promise.all (index 0)
at Commands.reconfigure (/opt/iobroker/node_modules/iobroker.zigbee/lib/commands.js:517:21))

**Comment by asgothian** [comment]
> hello, unfortunately there are no log entries with "elevated". I have now also updated the firmware of my Conbee2, but hasn't changed anything
> 
> I have set the IEEE in the debugmessage, i hope this is correct ... ![Screenshot 2023-02-19 114943](https://user-images.githubusercontent.com/62117738/219943504-dbabcbf9-63ac-410e-8211-c0d00b1b50c4.png) [iob_log.txt](https://github.com/ioBroker/ioBroker.zigbee/files/10776665/iob_log.txt)

The reason for not having no elevated messages is that you added the ieee with 0x in the state. If you remove the 0x, you should get messages.



> found this in the log....

Can you see if this message comes up again if you use the reconfigure button on the device tile ?

A.

**Comment by gemo-bi** [comment]
OK, now I can see the elevated messages.... and here are the humidity and temperatur
>>found this in the log.... yes this message comes on the reconfigureation.

Another point that I noticed, since the update of the conbee2, the signal strength is almost always 255.


2023-02-20 06:51:18.763	warn	ELEVATED publishToState: message received '{"battery":100}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:51:18.753	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:18.752	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:18.751	warn	ELEVATED publishToState: message received '{"linkquality":255}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:51:13.541	warn	ELEVATED publishToState: message received '{"humidity":44.5}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:51:13.536	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:13.536	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:13.535	warn	ELEVATED publishToState: message received '{"linkquality":255}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:51:08.350	warn	ELEVATED publishToState: message received '{"temperature":20.3}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:51:08.343	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:08.342	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:51:08.342	warn	ELEVATED publishToState: message received '{"linkquality":255}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:50:13.725	warn	ELEVATED publishToState: message received '{"battery":100}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:50:13.713	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:50:13.712	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:50:13.712	warn	ELEVATED publishToState: message received '{"linkquality":255}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:50:08.503	warn	ELEVATED publishToState: message received '{"humidity":44.5}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:50:08.496	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:50:08.496	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:50:08.495	warn	ELEVATED publishToState: message received '{"linkquality":255}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:49:08.704	warn	ELEVATED publishToState: message received '{"battery":100}' from device a4c13892e9c0e962 type 'TS0601_temperature_humidity_sensor'
2023-02-20 06:49:08.696	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'
2023-02-20 06:49:08.696	warn	ELEVATED publishToState: value generated '255' from device a4c13892e9c0e962 for 'Link quality'

**Comment by asgothian** [comment]
The device is definitely reporting all values. That is good. Now to figure out why there is no state attached.

Please try the following;

copy the following code and place it in a file placed in `/opt/iobroker/iobroker-data/zigbee_0`
I would suggest you call that file `ts0601_a.js`, but in the end, the filename does not matter

```
const exposes = require('../lib/exposes');
const fz = require('../converters/fromZigbee');
const tz = require('../converters/toZigbee');
const ota = require('../lib/ota');
const tuya = require('../lib/tuya');
const reporting = require('../lib/reporting');
const extend = require('../lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const libColor = require('../lib/color');
const utils = require('../lib/utils');
const globalStore = require('../lib/store');

module.exports = [
{
    fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_bq5c8xfe'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_bjawzodf'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_qyflbnbj'}],
    model: 'TS0601x_temperature_humidity_sensor',
    vendor: 'TuYa',
    description: 'Temperature & humidity sensor',
    fromZigbee: [fz.tuya_temperature_humidity_sensor],
    toZigbee: [],
    exposes: [e.temperature(), e.humidity(), e.battery()],
}
];

```

- go to the zigbee adapter configuration and enter the filename (without path, just the filename) into the external converter field.
- restart the adapter
- check the log for a message stating "external converter applied from ..."
- check the states and see if the behaviour changes.

A.

**Comment by gemo-bi** [comment]
That looks good.....now i can see the temperatur and the humidity, and also the objects are here.


![Screenshot 2023-02-20 085318](https://user-images.githubusercontent.com/62117738/220045958-128e37ff-731c-4bb1-900a-06fb7f2abb42.jpg)

**Comment by asgothian** [comment]
Ok.. you will need to run with this for a while - there seems to be an issue with how the exposes are handled in the original implementation. I doubt I can generate a quick fix easily.

Keeping the external converter has no negative effects for you (other than the missing image), and it will remain active through updates of zigbee adapter and zigbee-herdsman-converters libraries.

I will close this issue and open a new one with the technical explanations of what is going wrong.

**Comment by gemo-bi** [comment]
Thank you for your quick help. great job :)

**Comment by Dobbiiey** [comment]
Hello,
I had a similar issue with the TS0601 & struggeled long time as an ioBroker beginner.
Adding a new manufacturerName made it work ;-)
______________
const exposes = require('../lib/exposes');
const fz = require('../converters/fromZigbee');
const e = exposes.presets;

module.exports = [
{
    fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_bq5c8xfe'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_bjawzodf'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_qyflbnbj'},
        {modelID: 'TS0601', manufacturerName: '**_TZE200_zl1kmjqx**'}],
    model: 'TS0601x_temperature_humidity_sensor',
    vendor: 'TuYa',
    description: 'Temperature & humidity sensor',
    fromZigbee: [fz.tuya_temperature_humidity_sensor],
    toZigbee: [],
    exposes: [e.temperature(), e.humidity(), e.battery()],
}
];
______________

hope my comment helps someone

thanks to asgothian and all project supporter !
 
now only the humidety is a bit too high  :-(
![image](https://user-images.githubusercontent.com/124839322/226145785-089e70d2-8512-44c3-809a-05af84859e07.png)

**Comment by arteck** [comment]
install the GIT version and check again

**Comment by gemo-bi** [comment]
Now the temperatur and humidity is working. I have removed the ts0601_a.js file from the config.
thx to @arteck  :)
![Screenshot 2023-03-21 180723](https://user-images.githubusercontent.com/62117738/226687449-cf59b8cc-750c-467a-b49b-b0d17942100a.png)

**Comment by Dobbiiey** [comment]
@arteck
now it's fixed with git version v1.8.11! Humidity calculation works!
I still need the external converter with my "manufacturerName: '_TZE200_zl1kmjqx'"...
(and for sure: no warnings in logs value>100% every one minute)

![image](https://user-images.githubusercontent.com/124839322/226729082-3e825b32-ba6c-43ea-8b4d-20e04f58dabb.png)

![image](https://user-images.githubusercontent.com/124839322/226729179-a75176d1-5c3d-4d7c-a4f1-9698d1cd5414.png)

Thanks a lot !! I'm glad ;-) !

**Comment by daNick73** [comment]
Using this from @asgothian

https://github.com/ioBroker/ioBroker.zigbee/issues/1683#issuecomment-1436464323

I end up with the below... any recommendations?

`

host.raspberrypi | 2023-09-23 15:45:54.887 | info | Restart adapter system.adapter.zigbee.0 because enabled
-- | -- | -- | --
host.raspberrypi | 2023-09-23 15:45:54.886 | error | instance system.adapter.zigbee.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
host.raspberrypi | 2023-09-23 15:45:54.885 | error | Caught by controller[1]: at /opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11852:26
host.raspberrypi | 2023-09-23 15:45:54.884 | error | Caught by controller[1]: at Zigbee._callReadyHandler (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11875:18)
host.raspberrypi | 2023-09-23 15:45:54.883 | error | Caught by controller[1]: at Zigbee.emit (node:domain:489:12)
host.raspberrypi | 2023-09-23 15:45:54.882 | error | Caught by controller[1]: at Zigbee.emit (node:events:514:28)
host.raspberrypi | 2023-09-23 15:45:54.880 | error | Caught by controller[1]: at Zigbee.<anonymous> (/opt/iobroker/node_modules/iobroker.zigbee/main.js:67:37)
host.raspberrypi | 2023-09-23 15:45:54.855 | error | Caught by controller[1]: at Zigbee.onReady (/opt/iobroker/node_modules/iobroker.zigbee/main.js:191:14)
host.raspberrypi | 2023-09-23 15:45:54.854 | error | Caught by controller[1]: at Zigbee.applyExternalConverters (/opt/iobroker/node_modules/iobroker.zigbee/main.js:282:75)
host.raspberrypi | 2023-09-23 15:45:54.853 | error | Caught by controller[1]: TypeError: JSON.stringfy is not a function
host.raspberrypi | 2023-09-23 15:45:54.852 | error | Caught by controller[0]: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
zigbee.0 | 2023-09-23 15:45:52.634 | warn | Terminated (UNCAUGHT_EXCEPTION): Without reason
zigbee.0 | 2023-09-23 15:45:52.629 | info | terminating
zigbee.0 | 2023-09-23 15:45:52.620 | info | cleaned everything up...
zigbee.0 | 2023-09-23 15:45:52.563 | error | JSON.stringfy is not a function
zigbee.0 | 2023-09-23 15:45:52.560 | error | TypeError: JSON.stringfy is not a function at Zigbee.applyExternalConverters (/opt/iobroker/node_modules/iobroker.zigbee/main.js:282:75) at Zigbee.onReady (/opt/iobroker/node_modules/iobroker.zigbee/main.js:191:14) at Zigbee.<anonymous> (/opt/iobroker/node_modules/iobroker.zigbee/main.js:67:37) at Zigbee.emit (node:events:514:28) at Zigbee.emit (node:domain:489:12) at Zigbee._callReadyHandler (/opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11875:18) at /opt/iobroker/node_modules/@iobroker/js-controller-adapter/src/lib/adapter/adapter.ts:11852:26
zigbee.0 | 2023-09-23 15:45:51.178 | error | unhandled promise rejection: JSON.stringfy is not a function
zigbee.0 | 2023-09-23 15:45:51.174 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
zigbee.0 | 2023-09-23 15:45:51.143 | info | Apply converter from module: /opt/iobroker/iobroker-data/zigbee_0/ts0601_a.js
zigbee.0 | 2023-09-23 15:45:51.001 | info | starting. Version 1.8.24 (non-npm: ioBroker/ioBroker.zigbee) in /opt/iobroker/node_modules/iobroker.zigbee, node: v18.17.1, js-controller: 5.0.12

`
